sklearn/linear_model/_ransac.py

```diff
@@ -404,12 +423,37 @@ def fit(self, X, y, sample_weight=None):
         estimator_name = type(estimator).__name__
         if sample_weight is not None and not estimator_fit_has_sample_weight:
             raise ValueError(
-                "%s does not support sample_weight. Samples"
+                "%s does not support sample_weight. Sample"
                 " weights are only used for the calibration"
                 " itself." % estimator_name
             )
+
+        _raise_for_params(fit_params, self, "fit")
+
         if sample_weight is not None:
-            sample_weight = _check_sample_weight(sample_weight, X)
+            fit_params["sample_weight"] = sample_weight
+
+        if _routing_enabled():
+            try:
+                routed_params = process_routing(self, "fit", **fit_params)
+            except UnsetMetadataPassedError as e:
+                raise UnsetMetadataPassedError(
+                    message=(
+                        f"{e}, which is used internally by `RANSACRegressor.fit()`."
+                        f"Call `{estimator.__class__.__name__}.set_{{method}}_request("
+                        "{metadata}=True)` for each metadata."
+                    ),
```

Comment:
I mean, the whole message, not just a part of it.

Reply:
So you want me to provide how this error message would look like if a user would encounter it?
Sorry, seems I'm not getting your intention here.

Reply:
Yes, exactly. Cause I'm confused as why `e` is at the beginning of the message.

Reply:
An example would be:

sklearn.exceptions.UnsetMetadataPassedError: [sample_weight] are passed but are not explicitly set as requested or not for ConsumingRegressor.score, which is used internally by `RANSACRegressor.fit()`.Call `ConsumingRegressor.set_{method}_request({metadata}=True)` for each metadata.

Reply:
We have talked about that it makes sense to have this expanded error message for `RANSACRegressor`, because it internally uses `score` in `fit`. People would get a very confusing error otherwise, they would only see `e`, that would tell them to `set_score_request`, but they wouldn't know why.

> This is a pattern repeated pretty much everywhere when we call process_routing, do we want to change all of them with this pattern? 
This PR is about `RANSACRegressor` and I would not be in pro of changing any other error message unless there is a very good reason to do so.

What would be your suggestion of how to deal with it, @adrinjalali?